File: Data Structures & Algorithms/two-integer-sum-ii/submission-3.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        for i in range(len(numbers)):
            num = target - numbers[i]
            print(i+1, num,"i+1, num")
            if num + numbers[i] == target and num in numbers:
                print(num,"num")
                index = numbers.index(num)
                return[i+1, index+1]

            

File: Data Structures & Algorithms/two-integer-sum/submission-3.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) { 
        // for(let i = 0;i<nums.length; i++){
        //     for(let j = i+1;j<nums.length; j++){
        //         if(nums[i] + nums[j] === target){
        //             return [i, j]
        //         }
        //     }
        // }
        let numMap = new Map()

        for(let i = 0; i < nums.length; i++){ 
            let num = nums[i]
            let complement = target - num
            let subIndex = numMap.get(complement)
            let targetI = numMap.has(complement)
            if(targetI){
                return [i, subIndex]
            }
            numMap.set(num, i)
        }
        return [-1, -1]
    }
}
